plugin/src/main/java/hudson/plugins/swarm/PluginImpl.java

```diff
@@ -54,9 +57,28 @@ public void doCreateSlave(StaplerRequest req, StaplerResponse rsp, @QueryParamet
 				nodeProperties = Lists.newArrayList(new ToolLocationNodeProperty(parsedToolLocations));
             }
     		
-            // try to make the name unique. Swarm clients are often replicated VMs, and they may have the same name.
-            if (jenkins.getNode(name) != null) {
-                name = name + '-' + req.getRemoteAddr();
+            if (hash == null && jenkins.getNode(name) != null) {
```

Comment:
SwarmSlave is ephemeral so on restart none will exist.

Plus this branch is only if using a legacy client, wherein we can do
nothing as legacy cannot pick up the new name.

When a swarm save disconnects its node is removed. Should be fine

On Friday, April 24, 2015, James Nord <notifications@github.com> wrote:

> In plugin/src/main/java/hudson/plugins/swarm/PluginImpl.java
> <https://github.com/jenkinsci/swarm-plugin/pull/26#discussion_r29063891>:
>
> > @@ -54,9 +57,28 @@ public void doCreateSlave(StaplerRequest req, StaplerResponse rsp, @QueryParamet
> >  				nodeProperties = Lists.newArrayList(new ToolLocationNodeProperty(parsedToolLocations));
> >              }
> >      		
> > -            // try to make the name unique. Swarm clients are often replicated VMs, and they may have the same name.
> > -            if (jenkins.getNode(name) != null) {
> > -                name = name + '-' + req.getRemoteAddr();
> > +            if (hash == null && jenkins.getNode(name) != null) {
>
> does this mean that no existing swarm clients will be able to reconnect
> when Jenkins restarts - or when a swarm node disconnects its node is
> removed (and hence there would be no previous swamr nodes on a Jenkins
> restart?)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/swarm-plugin/pull/26/files#r29063891>.
>


-- 
Sent from my phone